src/elm/Page.elm

```diff
@@ -254,13 +254,17 @@ viewTitle text_ =
         [ text text_ ]
 
 
-viewHeader : LoggedIn.Model -> String -> Route -> Html msg
-viewHeader { shared } title route =
+viewHeader : LoggedIn.Model -> String -> Html msg
+viewHeader { shared, routeHistory } title =
     div [ class "w-full h-16 flex px-4 items-center bg-indigo-500" ]
         [ div [ class "flex container mx-auto" ]
             [ a
                 [ class "flex items-center mr-4"
-                , Route.href route
+                , routeHistory
+                    |> List.drop 1
+                    |> List.head
+                    |> Maybe.withDefault Route.Dashboard
+                    |> Route.href
```

Comment:
ohhhhh I never figured out how we would know this, dumb me it's simply a list 🤡

Reply:
To be fair, it took some trial and error to get it right and avoid any loops :laughing: